src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -536,6 +536,7 @@ export function useSwapInputsController({
 
   const onChangedPercentage = useDebouncedCallback(
     (percentage: number) => {
+      if (percentageToSwap.value === percentage) return;
```

Comment:
you unfortunately won't have guarantees that the `percentageToSwap` value will still be the "previous" value by the time this `onChangedPercentage` function is run (which is run via `runOnJS` which is async from the perspective of the UI thread it was called from)

you can see issues arise if you drop some logs and go from sliding to 1 to 0 and back and forth, the `percentageToSwap` value always ends up being equal to `percentage` 

separately, the `percentage` value (if not 0 or 1) is not directly comparable to the `percentageToSwap` value. The `percentageToSwap` value is a derived value and will "look nicer" than the exact `percentage` that is passed in.